src/shared/dissect-image.c

```diff
@@ -314,6 +314,17 @@ static int device_monitor_handler(sd_device_monitor *monitor, sd_device *device,
         return sd_event_exit(sd_device_monitor_get_event(monitor), r);
 }
 
+static int timeout_handler(sd_event_source *s, uint64_t usec, void *userdata) {
+        struct wait_data *w = userdata;
+        int r;
+
+        r = find_partition(w->parent_device, w->blkidp, &w->found);
+        if (r < 0)
```

Comment:
shouldn't this propagate all errors as they are, with the exception of ENXIO which should become ETIMEDOUT?

Reply:
and i think if the `r = sd_device_get_sysattr_value(d, "partition", &v);` line in `device_is_partition()` above fails with ENOENT because the sysfs attr file is missing, we should turns this into a recognizable error I think, and then also treat the same way. 

And the access could benefit from a comment, explaining the issue with the "partition" sysfs attribute